gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go

```diff
@@ -347,7 +354,11 @@ func (p *glusterBlockProvisioner) createVolume(volSizeInt int, blockVol string)
 			}
 			json.Unmarshal([]byte(out), execBlockRes)
 
-			if execBlockRes.User == "" || execBlockRes.AuthKey == "" {
+			if *execBlockRes == nil {
```

Comment:
check for json.Unmarshal error instead? it can be non-nil but some minor error (certain field didn't get filled in)